install-macos.sh

```diff
@@ -0,0 +1,139 @@
+#!/bin/bash
+
+set -euo pipefail
+
+readonly target_user="$1"
+readonly config="$2"
+readonly bucket="https://circleci-binary-releases.s3.amazonaws.com/circleci-launch-agent"
+
+prepare() {
+  echo "Checking needed commands are installed:"
+  command -v curl
+  command -v grep
+  command -v tee
+  command -v tar
+  command -v gzip
+  command -v sha256sum
+  echo
+
+  echo "Creating temp dir:"
+  #  declare tmp_dir
+  tmp_dir="$(mktemp -d -t runner-install)"
+  # shellcheck disable=SC2064
+  trap "rm -rf '${tmp_dir}'" EXIT
+  echo "$tmp_dir"
+  echo
+}
+
+prepare_user() {
+  echo "Setting up user '${1}':"
+
+  local workdir="/Users/${1}/workdir"
+  mkdir -p "${workdir}"
+  chmod 0750 "${workdir}"
+  chown "${1}:" "${workdir}"
+}
+
+download_agent() {
```

Comment:
how mac-specific is this bit?

I'm keen to keep our docs fairly agnostic, even if this makes things slightly more work for installers - as right now the set of customers we've had have all been pretty comfortable scripting up server configuration management

The reason the linux script is more manual than it needs to be is because I wanted people to walk through it and mostly understand what was going on and/or add it into their own provisioning scripts at a file-level

Reply:
The downloading might not be that specific, but all the other stuff is?